python/taichi/examples/simulation/laplace_equation.py

```diff
@@ -0,0 +1,307 @@
+        for j in range(maxElements):
+            if sources[j].q < 0 and ti.math.length(points[i] -
+                                                   sources[j].pos) < 0.025:
+                points[i] = vec2(-100, -100)
+            if dipoles[j].m != 0 and ti.math.length(points[i] -
+                                                    dipoles[j].pos) < 0.05:
+                points[i] = vec2(-100, -100)
+
+
+@ti.kernel
+def updateArrows():
+    for x, y in ti.ndrange(*arrowField):
+        vel = getVel(arrows[x, y].base)
+        arrows[x, y].vel = vel.norm()
+        arrows[x, y].dir = vel / vel.norm() / meshSpace / 1.5
+
+
+def drawArrows(_gui):
+    global fade
+    if fade < fadeMax:
+        fade += 1
+    if fade == 0:
+        updateArrows()  # after fade out, update arrow filed
+    else:
+        arr = arrows.to_numpy()
+        vel = arr['vel'].reshape(1, -1)[0]
+        vel = (vel / vel.max() * 0xdd + 0x11) * (math.fabs(fade / fadeMax))
+        mean = vel.mean()
+        if mean > 0x7f:
+            vel /= mean / 0x7f  # make uniform stream more beautiful
+        vel = vel.astype(int)
+        vel *= 2**16 + 2**8 + 1
+        _gui.arrows(
+            arr['base'].reshape(arr['base'].shape[0] * arr['base'].shape[1],
+                                2),
+            arr['dir'].reshape(arr['dir'].shape[0] * arr['dir'].shape[1], 2),
+            radius=1.5,
+            color=vel)
+
+
+def drawMark(_gui, _frame):
+    triangleTrans = [
+        vec2(0, 1) / guiHeight,
+        vec2(math.cos(7. / 6. * math.pi), math.sin(7. / 6. * math.pi)) /
+        guiHeight,
+        vec2(math.cos(-1. / 6. * math.pi), math.sin(-1. / 6. * math.pi)) /
+        guiHeight
+    ]
+    rectTrans = [
+        vec2(1 * screen[1] / screen[0], 1) / guiHeight,
+        vec2(-1 * screen[1] / screen[0], -1) / guiHeight,
+    ]
+    for i in range(maxElements):
+        if dipoles[i].m > 0:
+            _gui.circle(dipoles[i].pos, 0xFFFDC0, dipoles[i].m * 2000)
+        elif dipoles[i].m < 0:
+            _gui.circle(dipoles[i].pos, 0xD25565, dipoles[i].m * -2000)
+        if sources[i].q > 0:
+            _gui.rect(sources[i].pos + rectTrans[0] * 2 * sources[i].q,
+                      sources[i].pos + rectTrans[1] * 2 * sources[i].q,
+                      2 * sources[i].q + 1, 0xFFFDC0)
+        elif sources[i].q < 0:
+            _gui.rect(sources[i].pos + rectTrans[0] * 2 * sources[i].q,
+                      sources[i].pos + rectTrans[1] * 2 * sources[i].q,
+                      -2 * sources[i].q + 1, 0xD25565)
+        if vortexes[i].q != 0:
+            rotateMatrix = ti.Matrix(
+                [[
+                    math.cos(vortexes[i].q * dt * 40 * _frame),
+                    -math.sin(vortexes[i].q * dt * 40 * _frame)
+                ],
+                 [
+                     math.sin(vortexes[i].q * dt * 40 * _frame),
+                     math.cos(vortexes[i].q * dt * 40 * _frame)
+                 ]])
```

Comment:
```suggestion
            theta = vortexes[i].q * dt * 40 * _frame
            ct, st = ti.cos(theta), ti.sin(theta)
            rotateMatrix = ti.Matrix([[ct, -st], [st, ct]])
```
You are right. Then let's make the code more readable.